### ISSUE TITLE
Support all pod controllers

### ISSUE THREAD

**Issue by sudermanjr** [FairwindsOps/goldilocks#225]
This is a rollup of several issues. We should be able to support all pod controller types and be somewhat agnostic of what they are.

Issues included:
#217
#100 
#84 
#96

**Comment by rajakshay** [comment]
Is there an active plan for supporting this feature, or is this lower in priority at the moment, or blocked by refactoring of CLI summary?

**Comment by sudermanjr** [comment]
Unfortunately this is a fairly low priority at the moment as we have limited resources available to work on it. It will definitely be discussed in our Q1 planning in the next couple of weeks.

**Comment by ctml91** [comment]
Thanks for the update on this. I've been following this for a little bit now, I think the operator is great with the exception of this issue. The lack of support for all controllers VPA supports is preventing me from deploying the operator as it would only cover a fraction of workloads and I am not really interested in a partial solution. I have been exploring alternative solutions but am still hoping this makes it in sooner than.

**Comment by ninlil** [comment]
I've created a VPA for our StatefulSet:s on my own.. but they are not displayed in the dashboard...
If you have limited resources, could you please focus on enabling the dashboard to display any VPA, not just the one your controller creates?

**Comment by ctml91** [comment]
Is there any chance of this still being completed for the Q1 milestone (EOM) https://github.com/FairwindsOps/goldilocks/milestone/3? I still have some hope :)

**Comment by ctrought** [comment]
It would be nice to see this support DeploymentConfigs too if possible :) The [VPA in OpenShift](https://docs.openshift.com/container-platform/4.7/nodes/pods/nodes-pods-vertical-autoscaler.html) supports DeploymentConfigs.

**Comment by s7an-it** [comment]
@lucasreed @sudermanjr, when I deploy Argo Rollouts I get in controller 'no spec.template.spec field from unstructured workload' on the rollout object.", is that expected? Also in dashboards there are containers with no metrics.

**Comment by kolja-lucht** [comment]
I have a similar experience with the Rollouts. I installed Goldilocks on one of our EKS clusters today. The dashboard displays findings for Deployments, but not for Rollouts unfortunately. Here's a screenshot:
<img width="524" alt="Screenshot 2023-06-02 at 16 04 59" src="https://github.com/FairwindsOps/goldilocks/assets/6205839/bbedf106-4aa6-4d48-804b-b01271de0e92">

I used the Fairwinds Helm charts to install the VPA and Goldilocks (only the recommender) with the default values. Kubernetes version in the cluster is 1.23. Is there something I can change to get the Rollouts displayed as well?

**Comment by Eslam-mohammed-anwar** [comment]
This should solve the other issue for Rollout when it points to deployment
https://github.com/FairwindsOps/goldilocks/pull/685/files

**Comment by Yufeireal** [comment]
Hi team, we're seeing similar issues when using argocd Rollout points to Deployment. Do we still have this on our radar or it was fixed already, thanks!
